runtime/expr/function/ztests/grep.yaml

```diff
@@ -0,0 +1,16 @@
+# This test tests grep as a function call which only happens when the pattern
+# arg is not a glob, regular expression, or compile time resolvable to a string.
+
+script: |
+  echo '{pattern:"hello",input:[{a:{hello:"world"}},{hi:"world"}]}' \
+    | zq -z 'over input with p=pattern => ( grep(p) )' -
+  echo "// ==="
+  echo '{a:{foo:"bar"}} {b:{baz:"foo"}}' \
+    | zq -z 'func g(s, e): ( grep(s, e) ) where g("baz", this)' -
```

Comment:
```suggestion
  echo '{pattern:"hello",input:[{a:{hello:"world"}},{hi:"world"}]}' |
    zq -z 'over input with p=pattern => ( grep(p) )' -
  echo //
  echo '{a:{foo:"bar"}} {b:{baz:"foo"}}' |
    zq -z 'func g(s, e): ( grep(s, e) ) where g("baz", this)' -
```

Reply:
We typically use `// ===` to mark different tests, why change this now?

Reply:
`//` feels sufficient but I'm fine with `// ===`.